### PR TITLE
Option to reduce browser bundle size

### PR DESCRIPTION
**Pull request by waltjones**
fixes: https://github.com/rollbar/rollbar.js/issues/615

Allows the current standard package for browser to continue to work as is. Provides the ability for Webpack and other builders to use `require(‘rollbar/src/browser/core’)` instead. Just using this will build a minimal Rollbar.js with telemetry and other optional features not included. Each of these features can be included a la carte to provide the package needed at the smallest possible size.

Minimal example:
```
var Rollbar = require('rollbar/src/browser/core');
rollbar = Rollbar.init(config);
rollbar.log(‘Hello’);
```

Example with all optional features included:
```
var Rollbar = require('rollbar/src/browser/core');
var telemeter = require('rollbar/src/telemetry');
var instrumenter = require('rollbar/src/browser/telemetry');
var polyfillJSON = require('rollbar/vendor/JSON-js/json3');
var wrapGlobals = require('rollbar/src/browser/wrapGlobals');
var scrub = require('rollbar/src/scrub');
var truncation = require('rollbar/src/truncation');

Rollbar.setComponents({
  telemeter: telemeter,
  instrumenter: instrumenter,
  polyfillJSON: polyfillJSON,
  wrapGlobals: wrapGlobals,
  scrub: scrub,
  truncation: truncation
});
 
rollbar = Rollbar.init(config);
 
rollbar.log(‘Hello’);
```

The following size estimates are based on a vanilla webpack 4 config, and measure the pre-zip minified size.

The minified size of the minimal package (browser/core only) is 49K, down from 74K for a 34% reduction.

**Base telemeter** - The previously always-on telemetry capture for Rollbar message logging.
`require(rollbar/src/telemetry)` (saves ~3K minified)

**Browser telemetry** - All browser telemetry capabilities: Console, network, click events, etc.
`require(rollbar/src/browser/telemetry)` (saves ~14K minified)

**JSON polyfill** - Some pre-2009 browsers require a polyfill for JSON.stringify and JSON.parse.
`require(rollbar/vendor/JSON-js/json3)` (saves ~7K minified)

**Wrap globals** - This is already disabled by default. Most people would benefit from not bundling.
` require('rollbar/src/browser/wrapGlobals')` (saves ~1K)

**Scrub** - By making parts of the payload optional, there may be no remaining need for scrubbing in specific configurations. Or the remaining payload isn’t considered sensitive by the user. 
`require('rollbar/src/scrub')` (saves ~1K)

**Truncation** - Not likely needed when telemetry isn't present.  
`require('rollbar/src/truncation')` (saves ~1.6K)

**package.json** - Rollbar.js builds its /dist/ using webpack's `DefinePlugin` to load defaults from package.json. When building from /src/ though, all of package.json gets required. This PR refactors the defaults so it's not necessary to `require('package.json')`. (saves ~3K minified)

### For reviewers:
This is a large PR, but straightforward.

* Move the contents of browser/rollbar.js to browser/core.js. browser/rollbar.js is still the entry point for dist/rollbar.min.js. It now calls `Rollbar.setComponents()` with all optional components included.
* Because browser/rollbar.js uses `Rollbar.setComponents()`, all browser test automation now tests the new componentized build.
* Each component is extracted by lifting the require out of rollbar.js and into the app, then passing in as a `setComponent` option.
* Tests are added for the minimal browser/core.js build (with no optional components added.)

Each of the above is in its own commit, to make them easier to review in isolation.

### Future work
This PR does not create a minimal build target in dist (e.g. dist/rollbar.core.min.js). This can be added in the future if there's demand. The use cases we currently know about are focused around webpack builds, and similar build tools.

There's been discussion about having scoped packages (e.g. @rollbar/browser, @rollbar/core, etc.). This work is a logical precursor to that and can be used to generate those build targets.

### CODE REVIEW

File: src/api.js [hunk 1]
```diff
@@ -30,13 +29,14 @@ var defaultOptions = {
...
-  this.url = u;
-  this.jsonBackup = j;
+  this.transport = transport;
+  this.url = urllib;
+  this.truncation = truncation;
+  this.jsonBackup = jsonBackup;
   this.accessToken = options.accessToken;
-  this.transportOptions = _getTransport(options, u);
+  this.transportOptions = _getTransport(options, urllib);
 }
```
Comment (austenLacy) [inline src/api.js:40]: 🙌

File: src/browser/transforms.js [hunk 2]
```diff
@@ -309,11 +309,15 @@ function errorClass(stackInfo, guess, options) {
   }
 }
 
-function scrubPayload(item, options, callback) {
-  var scrubFields = options.scrubFields || [];
-  var scrubPaths = options.scrubPaths || [];
-  item.data = _.scrub(item.data, scrubFields, scrubPaths);
-  callback(null, item);
+function addScrubber(scrub) {
```
Comment (austenLacy) [inline src/browser/transforms.js:312]: not a big deal, but maybe `scrubFn` over `scrub` just to add some clarity

**Comment by FabianSellmann** [comment]
Will these modules be eventually included in the type definitions?

**Comment by waltjones** [comment]
@FabianSellmann Yes. https://github.com/rollbar/rollbar.js/issues/880

**Pull request merged**